lib/alchemy-pg_search.rb

```diff
@@ -41,12 +39,15 @@ def self.remove_page(page)
     def self.index_page(page)
       remove_page page
 
-      page.update_pg_search_document
+      document = page.update_pg_search_document
+      return if document.nil?
+      
       page.all_elements.includes(:ingredients).find_each do |element|
```

Comment:
Maybe we have an N+1 here

```rb
page.all_elements.includes(ingredients: {element: {page_version: :page}}).
```

Reply:
changed it and it is working without n+1s